src/Serializer/ItemNormalizer.php

```diff
@@ -37,11 +39,17 @@ public function normalize($object, $format = null, array $context = [])
 
     /**
      * {@inheritdoc}
+     *
+     * @throws InvalidArgumentException
      */
     public function denormalize($data, $class, $format = null, array $context = [])
     {
         // Avoid issues with proxies if we populated the object
         if (isset($data['id']) && !isset($context['object_to_populate'])) {
+            if (isset($context['allow_update']) && true !== $context['allow_update']) {
```

Comment:
Yes but again it's very unlikely.

When API Platform call the serializer's `serialize()` method (passing its custom keys) it's basically to pass them to its own normalizers (not Symfony one) handling only resources.

When another bundle will call the `serialize()` method, it will almost never trigger API Platform's serializer (the data must be a resource and so on).

Problems can happen when a property of a resource has a type that it not known by API Platform. In this case API Platform will delegate serialization of this data to the raw Symfony serializer. Here and only here conflicts can happen, but it's the very unlikely worst case, and anyway there are userland workarounds for that.

It doesn't worth it. We'll break too many existing apps, maybe introduce new bugs and make the config less readable.